### PR TITLE
Custom progress bar with rich/tqdm styles and config

### PR DESCRIPTION
**Pull request by adamgayoso**
1. scvi-specific progress bar with "rich" and "tqdm" style option.
2. `silent` option to silent the bar
3. ScviConfig to control global settings like this, random seed, and batch size after training to get quantities

Screen Shot 2020-09-12 at 3 06 31 PM

### CODE REVIEW

File: scvi/core/trainers/trainer.py [hunk 1]
```diff
@@ -173,11 +172,8 @@ def train(self, n_epochs=400, lr=1e-3, eps=0.01, params=None, **extras_kwargs):
 
         self.on_training_begin()
 
-        for self.epoch in tqdm(
-            range(n_epochs),
-            desc="training",
-            disable=not self.show_progbar,
-            file=sys.stdout,
+        for self.epoch in track(
+            range(n_epochs), description="Training...", disable=not self.show_progbar
```
Comment (galenxing) [inline scvi/core/trainers/trainer.py:176]: Thoughts on changing `self.show_progbar` to `self.silent`?

**Review by galenxing — COMMENTED** [review]
Looks good! Only this is when there are multiple functions that have rich output in the same cell in Jupyter, the newer output overwrites the previous output.

File: scvi/core/trainers/trainer.py (hunk 1 above)
Comment (adamgayoso) [reply scvi/core/trainers/trainer.py]: done

**Review by galenxing — APPROVED** [review]
LGTM

**Pull request merged**